### ISSUE TITLE
music metadata

### ISSUE THREAD

**Issue by roipoussiere** [tidalcycles/strudel#578]
Music examples provided in the REPL starts with some comments about the music, like:

```
// licensed with CC BY-NC-SA 4.0 https://creativecommons.org/licenses/by-nc-sa/4.0/
// by Felix Roos, inspired by Friendship - Let's not talk about it (1979) :)
```

It could be interesting to provide a standard for writing music metadata, such `@` followed by a tag (the syntax used in many doc generators):

```
// @license: CC BY-NC-SA 4.0 https://creativecommons.org/licenses/by-nc-sa/4.0/
// @author: Felix Roos
// @details: Inspired by Friendship - Let's not talk about it (1979) :)
```

This way, these information could be used by external tools, such as the [Strudel swatch](https://github.com/tidalcycles/strudel/tree/main/my-patterns) to display music title.

Available tags could be:
- `@title`: music title
- `@author`, `@authors`: music author(s), eventually followed with a link in `<>` (ex: `@author: john doe <https://example.com>`)
- `@license`: music license
- `@details`: some additional information about the music
- `@url`: web page related to the music (github repo, soundcloud link, etc.)
- `@genre`: music genre (pop, jazz, etc)
- `@album`: music album name

**Comment by felixroos** [comment]
another great idea :) also interesting for a potential future strudel / [dwitter](https://www.dwitter.net/) clone. 
The meta info could maybe even be autogenerated & edited with a form, to make sure the syntax is correct.

**Comment by roipoussiere** [comment]
> another great idea :)

Thanks! So can I make a PR to edit the header of the code samples licensed under CC-BY-NC-SA (yours, basically)?

> also interesting for a potential future strudel / [dwitter](https://www.dwitter.net/) clone.

Oh I didn't know dwitter, very fun!

I had a similar idea in mind: "Radio Strudel", which continuously plays various songs made by the Strudel community. Would love to work on it later. Also by loading 1mb of zipped code it could work for hours lol.

> The meta info could maybe even be autogenerated & edited with a form, to make sure the syntax is correct.

Yep totally agree!

**Comment by felixroos** [comment]
> I had a similar idea in mind: "Radio Strudel" ..

also a fun idea :)

> Thanks! If you want, I can make a PR to edit the header of the code samples licensed under CC-BY-NC-SA (yours, basically).

would be nice! Maybe leave out the colon, to have jsdoc syntax for the lines? like this:

```js
// @title ...
// @author ..., ..., ...
// @license ...
// @details ...
// @url ...
// @genre ...
// @album ...
```

I am not sure if if `author` is the right word.. mabye use `artist`, `name` or `by` ?

**Comment by roipoussiere** [comment]
> Maybe leave out the colon, to have jsdoc syntax for the lines?

Sure!

> I am not sure if if author is the right word.. mabye use artist, name or by ?

Indeed. I don't like `name` because it can be confused with the name of the song. I like `by`, it's short and clear.

**Comment by roipoussiere** [comment]
fixed in #580, closing
